include/pmacc/lockstep/Kernel.hpp

```diff
@@ -67,65 +67,185 @@ namespace pmacc::lockstep
+                    : m_UserKernelFunctor(kernelFunctor)
                     , m_metaData(file, line)
                 {
                 }
 
                 /** Configured kernel object.
-                 *
-                 * This objects contains the functor and the starting parameter.
                  *
                  * @tparam T_VectorGrid type which defines the grid extents
-                 * @tparam T_VectorBlock type which defines the block extents
+                 * @tparam T_blockSize number of block indices
                  *
-                 * @param gridExtent grid extent configuration for the kernel
-                 * @param blockExtent block extent configuration for the kernel
+                 * @param gridSize grid extent configuration for the kernel
                  *
                  * @return object to bind arguments to a kernel
                  *
                  * @{
                  */
+                template<uint32_t T_blockSize, typename T_VectorGrid>
+                HINLINE auto config(T_VectorGrid const& gridSize) const
+                {
+                    static_assert(
+                        lockstep::traits::hasBlockCfg_v<T_UserKernelFunctor> == false,
+                        "You are not allowed to overwrite the block size defined by the kernel!");
+                    auto blockCfg = lockstep::makeBlockCfg<T_blockSize>();
+                    using BlockConfiguration = decltype(blockCfg);
+                    constexpr uint32_t dim = pmacc::exec::detail::GetDim<T_VectorGrid>::dim;
+                    auto blockExtent = DataSpace<dim>::create(1);
+                    blockExtent.x() = BlockConfiguration::numWorkers();
+                    return pmacc::exec::detail::KernelLauncher<KernelFunctor<BlockConfiguration>, dim>{
+                        m_UserKernelFunctor,
+                        m_metaData,
+                        gridSize,
+                        blockExtent};
+                }
+
+                /**
+                 * The block size is derived from the kernel functor.
+                 * The kernel must define the member type `BlockDomSizeND` or the compile time constance
+                 * `blockDomSize`.
+                 */
                 template<typename T_VectorGrid>
-                HINLINE auto operator()(T_VectorGrid const& gridExtent) const
+                HINLINE auto config(T_VectorGrid const& gridSize) const
+                {
+                    static_assert(
+                        lockstep::traits::hasBlockCfg_v<T_UserKernelFunctor> == true,
+                        "Kernel must define the type BlockDomSizeND or the value blockDomSize!");
+                    auto blockCfg = lockstep::makeBlockCfg(m_UserKernelFunctor);
+                    using BlockConfiguration = decltype(blockCfg);
+                    constexpr uint32_t dim = pmacc::exec::detail::GetDim<T_VectorGrid>::dim;
+                    auto blockExtent = DataSpace<dim>::create(1);
```

Comment:
```suggestion
                    auto blockSize = DataSpace<dim>::create(1);
```
This seems like more consistent naming

Reply:
renamed it to `blockSizeND` to use the new naming schema